server/Arcadia.Assistant.Calendar.Abstractions/CalendarEvent.cs

```diff
@@ -16,13 +17,30 @@ public class CalendarEvent
 
         public string EmployeeId { get; }
 
-        public CalendarEvent(string eventId, string type, DatesPeriod dates, string status, string employeeId)
+        public DateTimeOffset CreateDate { get; }
+
+        public DateTimeOffset UpdateDate { get; }
+
+        public string UpdateEmployeeId { get; }
```

Comment:
I'm not happy about this line. You add some metadata in common class, and only CSP implementation knows how to deal with that. CSP logic depends on exact implementation of this field usage. It violates encapsulation for BOTH pieces. If you need, add multiple fields RejectedBy, CancelledBy, etc.